service/matching/task_queue_manager.go

```diff
@@ -371,12 +371,18 @@ func (c *taskQueueManagerImpl) SetInitializedError(err error) {
 	}
 }
 
-func (c *taskQueueManagerImpl) SetUserDataInitialFetch(err error) {
-	c.userDataInitialFetch.Set(struct{}{}, err)
-	if err != nil {
-		// We can't recover from here without starting over, so unload the whole task queue
-		c.lostOwnership.Store(true) // not really lost ownership but we want to skip the last write
-		c.unloadFromEngine()
+// Sets user data enabled/disabled and marks the future ready (if it's not ready yet).
+func (c *taskQueueManagerImpl) SetUserDataInitialFetch(disabledError error, futureError error) {
```

Comment:
True, it is a little confusing now. I can rename

Reply:
This is sort of a traditional callback, so something like `Got...` or `On...` makes more sense to me.

Reply:
Also, I really don't like that this accepts two errors. It's really unconventional, and it leads to statements like `SetUserDataInitialFetch(nil, nil)`, which are really hard to interpret. Usually, it's a result and an error obtaining that result. There's also enough branching here that I'm tempted to say we should just inline it because it seems like each callsite is doing something different.